selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/WorkspacesListTest.java

```diff
@@ -0,0 +1,367 @@
+
+/**
+ * @author Sergey Skorik
+ * @author Ihor Okhrimenko
+ */
+@Test(groups = TestGroup.OSIO)
+public class WorkspacesListTest {
+  private static final int EXPECTED_WORKSPACES_COUNT = 2;
+  private static final int BLANK_WS_MB = 2048;
+  private static final int JAVA_WS_MB = 3072;
+  private static final int BLANK_WS_PROJECTS_COUNT = 0;
+  private static final int JAVA_WS_PROJECTS_COUNT = 1;
+  private static final String EXPECTED_DOCUMENTATION_PAGE_TITLE = "What Is a Che Workspace?";
+  private static final String EXPECTED_JAVA_PROJECT_NAME = "web-java-spring";
+  private static final String NEWEST_CREATED_WORKSPACE_NAME = "just-created-workspace";
+
+  @Inject private Dashboard dashboard;
+  @Inject private WorkspaceProjects workspaceProjects;
+  @Inject private WorkspaceConfig workspaceConfig;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private Workspaces workspaces;
+  @Inject private NewWorkspace newWorkspace;
+  @Inject private TestWorkspaceProvider testWorkspaceProvider;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private TestWorkspaceServiceClient testWorkspaceServiceClient;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private DocumentationPage documentationPage;
+  @Inject private WorkspaceOverview workspaceOverview;
+
+  private TestWorkspace blankWorkspace;
+  private TestWorkspace javaWorkspace;
+  private Workspaces.WorkspaceListItem expectedBlankItem;
+  private Workspaces.WorkspaceListItem expectedJavaItem;
+  private Workspaces.WorkspaceListItem expectedNewestWorkspaceItem;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/defaultSpringProjectWithDifferentTypeOfFiles");
+
+    blankWorkspace = testWorkspaceProvider.createWorkspace(defaultTestUser, 2, DEFAULT);
+    javaWorkspace = testWorkspaceProvider.createWorkspace(defaultTestUser, 3, UBUNTU_JDK8);
+
+    blankWorkspace.await();
+    javaWorkspace.await();
+
+    testProjectServiceClient.importProject(
+        javaWorkspace.getId(), Paths.get(resource.toURI()), "web-java-spring", MAVEN_SPRING);
+
+    expectedBlankItem =
+        new Workspaces.WorkspaceListItem(
+            defaultTestUser.getName(),
+            blankWorkspace.getName(),
+            BLANK_WS_MB,
+            BLANK_WS_PROJECTS_COUNT);
+    expectedJavaItem =
+        new Workspaces.WorkspaceListItem(
+            defaultTestUser.getName(), javaWorkspace.getName(), JAVA_WS_MB, JAVA_WS_PROJECTS_COUNT);
+
+    expectedNewestWorkspaceItem =
+        new Workspaces.WorkspaceListItem(
+            defaultTestUser.getName(),
+            NEWEST_CREATED_WORKSPACE_NAME,
+            BLANK_WS_MB,
+            BLANK_WS_PROJECTS_COUNT);
+
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    deleteIfWorkspaceExist(expectedJavaItem.getWorkspaceName());
```

Comment:
We doesn't need to do it if test workspace is been injected.

Reply:
after using @InjectTestWorkspace we still have one workspace which is created by UI, so it should be deleted in tearDown() method.